tensorflow_io/dicom/kernels/decode_dicom_image.cc

```diff
@@ -0,0 +1,293 @@
+    }
+
+    ~DecodeDICOMImageOp()
+    {
+        DcmRLEDecoderRegistration::cleanup(); // deregister RLE codecs
+        DJDecoderRegistration::cleanup();     // deregister JPEG codecs
+        DJLSDecoderRegistration::cleanup();   // deregister JPEG-LS codecs
+    }
+
+    void Compute(OpKernelContext *context) override
+    {
+        // Grab the input file content tensor
+        const Tensor &in_contents = context->input(0);
+        OP_REQUIRES(context, TensorShapeUtils::IsScalar(in_contents.shape()),
+                    errors::InvalidArgument(
+                        "DecodeDICOMImage expects input content tensor to be scalar, but had shape: ",
+                        in_contents.shape().DebugString()));
+
+        const auto in_contents_scalar = in_contents.scalar<string>()();
+
+        // Load Dicom Image
+        DcmInputBufferStream dataBuf;
+        dataBuf.setBuffer(in_contents_scalar.data(), in_contents_scalar.length());
+        dataBuf.setEos();
+
+        DcmFileFormat *dfile = new DcmFileFormat();
+        dfile->transferInit();
+        OFCondition cond = dfile->read(dataBuf);
+        dfile->transferEnd();
+
+        DicomImage *image = NULL;
+        try
+        {
+            image = new DicomImage(dfile, EXS_Unknown, CIF_DecompressCompletePixelData);
+        }
+        catch (...)
+        {
+            image = NULL;
+        }
+
+        unsigned long frameWidth = 0;
+        unsigned long frameHeight = 0;
+        unsigned int pixelDepth = 0;
+        unsigned long dataSize = 0;
+        unsigned long frameCount = 0;
+        unsigned int samples_per_pixel = 0;
+
+        if ((image == NULL) || (image->getStatus() != EIS_Normal))
+        {
+            if (on_error == "strict")
+            {
+                OP_REQUIRES(context, false,
+                            errors::InvalidArgument("Error loading image"));
+                return;
+            }
+            else if ((on_error == "skip") || (on_error == "lossy"))
+            {
+                Tensor *output_tensor = NULL;
+                OP_REQUIRES_OK(context, context->allocate_output(0,
+                                                                 {0},
+                                                                 &output_tensor));
+                return;
+            }
+        }
+
+        // Get image information
+        frameCount = image->getFrameCount(); // getNumberOfFrames(); starts at version DCMTK-3.6.1_20140617
+        frameWidth = image->getWidth();
+        frameHeight = image->getHeight();
+        pixelDepth = image->getDepth();
+        samples_per_pixel = image->isMonochrome() ? 1 : 3;
+
+        // Create an output tensor shape
+        TensorShape out_shape;
+        if ((samples_per_pixel == 1) && (color_dim == false))
+        {
+            out_shape = TensorShape({static_cast<int64>(frameCount), static_cast<int64>(frameHeight), static_cast<int64>(frameWidth)});
```

Comment:
The line length needs to be less than 80 chars. You can check the C++ code style using `clang-format`. More details can be found [here](https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md#c-coding-style).